### PR TITLE
add strongErrorHandler configuration

### PR DESCRIPTION
**Pull request by loay**

### CODE REVIEW

**Comment by loay** [comment]
@bajtos PTAL
related workspace PR: https://github.com/strongloop/loopback-workspace/pull/276

File: lib/jsonrpc-adapter.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ JsonRpcAdapter.prototype.getRoutes = function(obj) {
   return routes;
 };
 
-JsonRpcAdapter.errorHandler = function() {
+JsonRpcAdapter.strongErrorHandler = function() {
```
Comment (bajtos) [inline lib/jsonrpc-adapter.js]: What is the purpose of this API rename? Why can't we keep existing `errorHandler` name?

File: lib/rest-adapter.js [hunk 2]
```diff
@@ -341,7 +341,7 @@ RestAdapter.urlNotFoundHandler = function() {
   };
 };
 
-RestAdapter.errorHandler = function(options) {
+RestAdapter.strongErrorHandler = function(options) {
```
Comment (bajtos) [inline lib/rest-adapter.js]: We need to change the implementation to use the new strong-error-handler module under the hood, the method names (API) can stay the same, no need to change that.

File: test/rest.test.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ describe('strong-remoting-rest', function() {
       process.env.NODE_ENV = 'test';
     }
     objects = RemoteObjects.create({ json: { limit: '1kb' },
-      errorHandler: { disableStackTrace: false }});
+      strongErrorHandler: { debug: true }});
```
Comment (bajtos) [inline test/rest.test.js]: Ideally, we should preserve backwards compatibility in the sense that it is still possible to configure the error handler via `errorHandler: { disableStackTrace: false }}`.

Note that the fact that we use strong-error-handler under the hood is just an implementation detail that API consumers should not need to know about.

File: test/rest.test.js [hunk 4]
```diff
@@ -110,7 +110,7 @@ describe('strong-remoting-rest', function() {
         }
       );
 
-      objects.options.errorHandler.handler = function(err, req, res, next) {
+      objects.options.strongErrorHandler.handler = function(err, req, res, next) {
```
Comment (bajtos) [inline test/rest.test.js]: Ditto, this line should not need to be changed, the old API should continue to work in the new version.

File: test/rest.test.js [hunk 5]
```diff
@@ -127,7 +127,7 @@ describe('strong-remoting-rest', function() {
     });
 
     it('should disable stack trace', function(done) {
-      objects.options.errorHandler.disableStackTrace = true;
+      objects.options.strongErrorHandler.debug = false;
```
Comment (bajtos) [inline test/rest.test.js]: Ditto, this line should not need to be changed, the old API should continue to work in the new version.

If you prefer, then we can support both settings `debug:false` and `disableStackTrace:true`, but it's important to ensure that `disableStackTrace` keeps working.

However, while you are making this change, please change the default value of `disableStackTrace` to `true`, so that apps are configured for security by default.

File: test/rest.test.js [hunk 6]
```diff
@@ -2227,7 +2227,6 @@ describe('strong-remoting-rest', function() {
         for (var prop in expected) {
           expect(result.body.error[prop], prop).to.equal(expected[prop]);
         }
-        expect(result.body.error.stack, 'stack').to.contain(__filename);
```
Comment (bajtos) [inline test/rest.test.js]: Why is this change needed? I don't mind much, I just want to ensure we fully understand why we are making it.

**Comment by bajtos** [comment]
@loay The idea is to rework strong-remoting to utilize the new strong-error-handler under the hood, so that we can get the benefit of higher security, while preserving public API as much as we can.

**Comment by bajtos** [comment]
@loay @davidcheung PTAL please, I think this is no longer relevant, in which case we should close it.

**Comment by davidcheung** [comment]
this should be covered by https://github.com/strongloop/strong-remoting/pull/302
going to close this PR 
cc @loay
